tests/integration/eventing/userDefinedMessage.ts

```diff
@@ -0,0 +1,162 @@
+import * as assert from 'assert';
+import axios from 'axios';
+import Device from '../../../lib/twilio/device';
+import type Call from '../../../lib/twilio/call';
+import { generateAccessToken } from '../../lib/token';
+const env = require('../../env');
+
+const RELAY_SERVER_URL = 'http://localhost:3030';
+
+describe('userDefinedMessage', function() {
+  this.timeout(1000 * 60 * 10); // 10 minute timeout for the whole suite
+
+  const createCallTest = async () => {
+    const tokenTtl = 60 * 3; // 3 minute TTL
+
+    const aliceId = `client-id-eventing-tests-alice-${Date.now()}`;
+    const aliceToken = generateAccessToken(aliceId, tokenTtl, env.appSid);
+    const aliceDevice = new Device(aliceToken);
+
+    const bobId = `client-id-eventing-tests-bob-${Date.now()}`;
+    const bobToken = generateAccessToken(bobId, tokenTtl, env.appSid);
+    const bobDevice = new Device(bobToken);
+
+    await bobDevice.register();
+    const bobCallPromise = new Promise<Call>((res) => {
+      bobDevice.on(Device.EventName.Incoming, (c) => res(c));
+    });
+
+    const aliceCall = await aliceDevice.connect({ params: { To: bobId } });
+    const bobCall = await bobCallPromise;
+
+    const aliceCallAcceptPromise = new Promise((res) => {
+      aliceCall.on('accept', () => {
+        res();
+      });
+    });
+    const bobCallAcceptPromise = new Promise((res) => {
+      bobCall.on('accept', () => {
+        res();
+      });
+    });
+
+    bobCall.accept();
+
+    await aliceCallAcceptPromise;
+    await bobCallAcceptPromise;
+
+    const performTeardown = () => {
+      aliceDevice.destroy();
+      bobDevice.destroy();
+    };
+
+    return {
+      performTeardown,
+      alice: {
+        device: aliceDevice,
+        call: aliceCall,
+      },
+      bob: {
+        device: bobDevice,
+        call: bobCall,
+      },
+    };
+  };
+
+  describe('outgoing call leg', function() {
+    it(
+      'should successfully send a message to the customer server',
+      async function() {
+        const { performTeardown, alice: { call } } = await createCallTest();
```

Comment:
Can we move this in a `beforeEach` so we don't duplicate for every test? And `performTeardown` should be in an `afterEach`.